### PR TITLE
Added Spanish

### PR DESCRIPTION
**Pull request by deltazerorsan**
The main interface, as planned by David, can now be charged in English or Spanish.

### CODE REVIEW

**Comment by deltazerorsan** [comment]
This is the new approach. Maybe there is no need to close the previous pull...

File: Prefill_ES.ino [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  TXT_BFS8            = "Vendaval";
+  TXT_BFS9            = "Vendaval severo";
+  TXT_BFS10           = "Tormenta";
+  TXT_BFS11           = "Tormenta violenta";
+  TXT_BFS12           = "Huracán";
+//Days of the week
+  TXT_SUN             = "Dom";
+  TXT_MON             = "Lun";
+  TXT_TUE             = "Mar";
+  TXT_WED             = "Mie";
```
Comment (jsvaliente) [inline Prefill_ES.ino]: Missing accent. It should be 'Mié'.

File: Prefill_ES.ino [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  TXT_THU             = "Jue";
+  TXT_FRI             = "Vie";
+  TXT_SAT             = "Sab";
```
Comment (jsvaliente) [inline Prefill_ES.ino]: Missing accent. It should be 'Sáb'.

File: Prefill_ES.ino [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  Hemisphere          = "north";
+  Units               = "M";
+  TXT_FORECAST_GRAPHS = "Pronóstico a 5 días / 3 horas";
+  TXT_TEMPDEWPOINT    = "Temperatura / Punto de rocío";
+  TXT_TEMPERATURE     = "Temperatura";
+  TXT_FEELSLIKE       = "Sensación";
+  TXT_HUMIDITY        = "Humedad";
+  TXT_DEWPOINT        = "Punto de rocío";
+  TXT_WINDCHILL       = "Sensación térmica";
+  TXT_WCHILL_HINDEX   = "Sensación térmica / Indice de calor";
```
Comment (jsvaliente) [inline Prefill_ES.ino]: Missing accent. It should be 'Sensación térmica / Índice de calor'.

**Comment by deltazerorsan** [comment]
Thanks Javier.

¿Podrías revisar la traducción de "Feelslike" y "Windchill"? ¿Tenemos términos en español diferentes para esos en Inglés?

(Could you check the translation of "Feelslike" and "Windchill"? Do we have different Spanish terms for those in English?)

**Comment by G6EJD** [comment]
1. You need to remove prefill_EN() from sketch.
2. Rename prefill_ES.ino to prefill_ES.**h**
3. Add at the start #include "prefill_ES.h"
4. Make sure prefill_ES.h is in the sketch folder

**Comment by deltazerorsan** [comment]
Hi Dave,

The .h approach is not easy because the compiler need to know the functions first before using them.

I see not very functional to declare prototypes to solve that so at last my solution is to use conditional compilation.

It works, but tell me if you hate that solution.

Roman

**Review by G6EJD — CHANGES_REQUESTED** [review]
The best way to do this is use #include statements.
1. ESP32_OWM...ino 
2. Prefill_EN.h
3. Prefill_ES.h
4. Prefill_DE.h. And so on with others

Now just add the include as required:
//#include “prefill_EN.h
#include “prefill_ES.h”

And remove the prefill_EN() function from the main source code
What about the html menu items still in EN ?

**Comment by deltazerorsan** [comment]
Hi Dave,

The conversion to .h is not straight forward because you break the rule "first declare, then use" due that the Prefill function call another three functions from inside the function it self. That can be solved declaring prototypes before the include statement but that does not solve the Help page...

I'm looking for another approach...

Yours, Roman

**Comment by jsvaliente** [comment]
Shouldn't the .h file corresponding to the pre-fill match somehow the "String Language” variable from “own_credentials.h” (or viceversa). Just sayin’….

**Comment by deltazerorsan** [comment]
> Shouldn't the .h file corresponding to the pre-fill match somehow the "String Language” variable from “own_credentials.h” (or viceversa). Just sayin’….

Yes, not bad idea, but first things first.

Really the include statement can be inside own_credentials.h so there is only one place to touch...

**Comment by G6EJD** [comment]
Try this one:
https://github.com/G6EJD/ESP32_Advanced_Weather_Server V3

you need ‘credentials_EN.h’

**Comment by deltazerorsan** [comment]
This is. Take a look at the screenshot.

**Comment by G6EJD** [comment]
There is too much change for me to see through this, I can approve credentials_es.h

**Comment by deltazerorsan** [comment]
I'll try to simplify this ... I can't promise anything.

**Comment by deltazerorsan** [comment]
Dave,

Please try to merge now. If in doubt I can close this pull an start over...

My apologizes for the inconveniences.

Roman

**Comment by G6EJD** [comment]
That worked, thank you for doing this

**Comment by deltazerorsan** [comment]
A pleasure!

**Pull request merged**